lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -642,6 +680,330 @@ def get_fq_stats(self, reads_object, file_path):
+        self._download_direct_download_link(force_download_link, copy_file_path)
+
+    def _download_ftp_link(self, file_url, copy_file_path):
+        """
+        _download_ftp_link: FTP download link handler
+                            URL fomat: ftp://anonymous:email@ftp_link 
+                                    or ftp://ftp_link
+                            defualt user_name: 'anonymous'
+                                    password: 'anonymous@domain.com'
+
+                            Note: Currenlty we only support anonymous FTP due to securty reasons.
+
+        params:
+        file_url: FTP download link
+        copy_file_path: output file saving path
+
+        """
+        print "=== WARNING ===\n=== WARNING ===\n"
+        print "FTP IS NOT A SECURIED PROTOCOL\nCREDENTIALS AND FILE CONTENTS WILL BE EXPOSED TO THE PUBLIC"
+        print "\n=== STOP THE PROCESS NOW IF YOU DO NOT WISH TO EXPOSE PRIVATE DATA ==="
+        time.sleep(60)
+
+        self.log('Connecting FTP link: %s' % file_url)
+        ftp_url_format = re.match(r'ftp://.*:.*@.*/.*', file_url)
+        # process ftp credentials
+        if ftp_url_format:
+            self.ftp_user_name = re.search('ftp://(.+?):', file_url).group(1)
+            if self.ftp_user_name.lower() != 'anonymous':
+                raise ValueError("Currently we only support anonymous FTP")
+            self.ftp_password = file_url.rpartition('@')[0].rpartition(':')[-1]
+            self.ftp_domain = re.search('ftp://.*:.*@(.+?)/', file_url).group(1)
+            self.ftp_file_path = file_url.partition('ftp://')[-1].partition('/')[-1].rpartition('/')[0]
+            self.ftp_file_name = re.search('ftp://.*:.*@.*/(.+$)', file_url).group(1)
+        else:
+            self.log('Setting anonymous FTP user_name and password')
+            self.ftp_user_name = 'anonymous'
+            self.ftp_password = 'anonymous@domain.com'
+            self.ftp_domain = re.search('ftp://(.+?)/', file_url).group(1)
+            self.ftp_file_path = file_url.partition('ftp://')[-1].partition('/')[-1].rpartition('/')[0]
+            self.ftp_file_name = re.search('ftp://.*/(.+$)', file_url).group(1)
+
+        self._check_ftp_connection(self.ftp_user_name, self.ftp_password, 
+                                    self.ftp_domain, self.ftp_file_path, self.ftp_file_name)
+        
+        ftp_connection = ftplib.FTP(self.ftp_domain)
+        ftp_connection.login(self.ftp_user_name, self.ftp_password)
+        ftp_connection.cwd(self.ftp_file_path)
+
+        ftp_copy_file_path = copy_file_path + '.gz' if self.ftp_file_name.endswith('.gz') else copy_file_path
+        with open(ftp_copy_file_path, 'wb') as output:
+            ftp_connection.retrbinary('RETR %s' % self.ftp_file_name, output.write)
+        self.log('Copied FTP file to: %s' % ftp_copy_file_path)
+
+        if self.ftp_file_name.endswith('.gz'):
+            self._unpack_gz_file(copy_file_path)
```

Comment:
yup yup. It's a very very temp thing for now. I am not happy this _unpack_gz_file either.